src/main/scala/mesosphere/marathon/api/akkahttp/SystemController.scala

```diff
@@ -20,23 +23,40 @@ import scala.concurrent.ExecutionContext
 /**
   * The SystemController handles system level functionality like configuration, metrics and logging.
   */
-class SystemController(cfg: Config)(
+class SystemController(val marathonConfig: MarathonConf, val cfg: Config, val electionService: ElectionService)(
     implicit
     val actorSystem: ActorSystem,
     val executionContext: ExecutionContext,
     val authenticator: Authenticator,
-    val authorizer: Authorizer,
-    val electionService: ElectionService
-) extends Controller with StrictLogging {
+    val authorizer: Authorizer
+) extends Controller with MetricsConversion with StrictLogging {
 
   import Directives._
   import EntityMarshallers._
 
+  def acceptsTextPlain(ranges: Seq[MediaRange]): Boolean = {
```

Comment:
Do you think we can build a `Unmarshaller which will handle all those types instead of matching them by our own?